include/caffe/util/external_lib_data_source_util.hpp

```diff
@@ -0,0 +1,30 @@
+/**
+* @brief    This file contains helper interface declarations for external
+*           library data source.
+*/
+#ifndef EXTERNAL_LIB_DATA_SOURCE_UTIL_H_
+#define EXTERNAL_LIB_DATA_SOURCE_UTIL_H_
+
+#include <boost/shared_ptr.hpp>
+#include <string>
+
+class IExternalLibDataSource;
+
+/**
+* @brief    Class that abstracts away platform dependent library handling.
+*/
+class IExternalLib {
+ public:
+  virtual ~IExternalLib() {}
+  virtual IExternalLibDataSource* GetDataSource() = 0;
```

Comment:
It would be nice if the class for loading symbols/functions from a library could be generalized (possibly through templating) to support loading functions with any signature. Then it could be reused for loading compute layers as well.